### PR TITLE
portfolio: build from the sources

### PR DESCRIPTION
**Pull request by LeSuisse**
## Description of changes

Package is now built from the sources directly using `buildMavenPackage`. It should also work on aarch64-linux.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/office/portfolio/default.nix [hunk 1]
```diff
@@ -25,15 +26,21 @@ let
...
-    url = "https://github.com/buchen/portfolio/releases/download/${version}/PortfolioPerformance-${version}-linux.gtk.x86_64.tar.gz";
-    hash = "sha256-bm38t8wgLEYZVDsMnOv3Wj6TgKu2s36wo8q0Sj7pPNI=";
+  src = fetchFromGitHub {
+    owner = "buchen";
+    repo = "portfolio";
+    rev = version;
+    hash = "sha256-bZCWjJ0+yDu/oGGutebh5ZSaCF3lyn/ghSo73mpM8Po=";
   };
 
+  mvnHash = "sha256-gbb/lhobNlAHggWIiAuPbQMYBak3b8F0mvLNEJ7sVOk=";
```
Comment (kirillrdy) [inline pkgs/applications/office/portfolio/default.nix:40]: hash missmatch

Comment (kilianar) [reply pkgs/applications/office/portfolio/default.nix:40]: I've executed the build twice and obtained two distinct hashes for the Maven dependencies. If I recall correctly, I previously attempted to build portfolio from its source some time ago but abandoned the endeavor when I encountered difficulty obtaining a consistent hash.

Comment (LeSuisse) [reply pkgs/applications/office/portfolio/default.nix:40]: Indeed, sad. I'm going to close the PR, it was an opportunistic PR and I do not have the motivation to deal with some Maven shenanigan

**Comment by LeSuisse** [comment]
If someone want to get another stab at this you might want to take a look at #278788. I'm guessing some of the tricks could be re-used for `portfolio`.

**Comment by TomaSajt** [comment]
Okay, so I took a quick look at it, and maybe the problem is with tycho. There's a similar issue with `dbeaver` https://github.com/NixOS/nixpkgs/issues/267405#issuecomment-1881782604
Not 100% sure yet, but I'll do a bit more experimenting

**Pull request closed without merging**